### ISSUE TITLE
`update:modelValue` not called anymore on Taginput

### ISSUE THREAD

**Issue by Massedil** [oruga-ui/oruga#967]
### Overview of the problem

**Oruga** version: from 0.8.7 to 0.8.11
**Vuejs** version: 3.4.20
**OS/Browser**: Firefox

### Description

Hello,

We use a `<o-taginput>`, like this :

```
<o-taginput
  :modelValue="tagsStrings"
  @update:modelValue="updateTags"
  [...]
>
</o-taginput>
```

And the `updateTags` function is here :

```
const updateTags = (newTagsStrings: string[]) => {
  const tagEntities = newTagsStrings.map((tag: string | ITag) => {
    if (typeof tag !== "string") {
      return tag;
    }
    return { title: tag, slug: tag } as ITag;
  });
  emit("update:modelValue", tagEntities);
};

```

Since oruga-next 0.8.7, the `updateTags` function is not called anymore.

Here is what I have tested :

- 0.8.4 => bug datepicker, inusable
- 0.8.5 => `@update:modelValue` works
- 0.8.6 => `@update:modelValue`  works
- 0.8.7 => `@update:modelValue`  DONT works from here
- [not tested]
- 0.8.10 => `@update:modelValue`   still DONT works
- 0.8.11 => inusable (ReferenceError: can't access lexical declaration 'isMobileNative' before initialization)

Here is original code :
- https://framagit.org/framasoft/mobilizon/-/blob/main/src/components/Event/TagInput.vue
- https://framagit.org/framasoft/mobilizon/-/blob/1217f8a7779c6c3b25007dada336d5757cf358fd/src/components/Event/TagInput.vue

To temporarily solve the problem, I use the `@remove` and `@add` event to force call `updateTags` :
- https://framagit.org/massedil/mobilizon/-/commit/e6ab2147d9bf2f7dfe72b81964bea097dd4886c5

Is it normal that `@update:modelValue` stopped working for us ?

**Comment by mlmoravek** [comment]
Hi @Massedil, `@update:modelValue` should always work when you have a `modelValue` prop. 
We are slowly switching from our custom bidirectional v-model binding helper to the vue native `defineModel` implementation. 

I have tested the taginput component on the current dev and master branches and it seems to work for me.

**Comment by Massedil** [comment]
Hi ! Thanks for your quick reply !

So should I wait for a new release to test it or you have seen something wrong in our code ?

Could you reproduce the problem in 0.8.10 ? And make it works just by downgrading to 0.8.6 ?

**Comment by mlmoravek** [comment]
Sorry I can't repduce it. 

How is your `tagsStrings` instantiated?

**Comment by Massedil** [comment]
> How is your tagsStrings instantiated? Have you instantiated tagsStrings with an empty array like ref([]) ?

```
const tagsStrings = ref<string[]>([]);
```

**Comment by mlmoravek** [comment]
I can't reproduce it myself. 
Could you make a reproducible example in a playground? Otherwise, I'm sorry, I can't help you :/

**Comment by Massedil** [comment]
This is not working for me in 0.8.10 and working in 0.8.6 : 

```
<template>
  <p style="color: red">{{ tagsStrings }}</p>
  <o-taginput v-model="tagsStrings" @update:modelValue="updateTags" />
</template>
<script lang="ts" setup>
import { ref } from "vue";

const tagsStrings = ref<string[]>([]);

const updateTags = (newTagsStrings: string[]) => {
  alert("called");
  console.error("updateTags" + newTagsStrings[0]);
};
</script>
```

**Comment by Massedil** [comment]
> Could you make a reproducible example in a playground?

Do you have an empty playground example with just oruga-ui loaded ?

**Comment by mlmoravek** [comment]
You can makeyour own by just fork from the vue template and add Oruga. 

My example works for me with Oruga 0.8.10:
https://codesandbox.io/p/devbox/infallible-dust-9j4xrr?workspaceId=e57f0cf5-b7a7-4a98-914b-df34864c91af

**Comment by mlmoravek** [comment]
already tried to remove node_module and install all dependencies again?

**Comment by Massedil** [comment]
Is it helpful if I run the Mobilizon code on a server for you to test ?

I created a branch here : 
https://framagit.org/massedil/mobilizon/-/commits/update-modelValue-not-called/?ref_type=heads

Here you can see the simplified code that works on 0.8.6 : 
https://framagit.org/massedil/mobilizon/-/commit/457f558c18e821016307cce0be6c6a22e8c9c1c1

Here the only change I do to make the `@update:modelValue="updateTags"` not working is going to 0.8.10 : 
https://framagit.org/massedil/mobilizon/-/commit/a9286e0c3653e2b2bc06489e6851cee80e71be25

**Comment by Massedil** [comment]
> already tried to remove node_module and install all dependencies again?

Yes. To be sure I have done it again just now. Problem is still here.

**Comment by Massedil** [comment]
And here is my changes to emulate `@update:modelValue="updateTags"` on 0.8.10 : 
https://framagit.org/massedil/mobilizon/-/commit/4024ccddd0faf1c9bd98e1647faad698da9068da

**Comment by mlmoravek** [comment]
I saw that you have an old "@vue/compiler-sfc" also integrated with a different version of your vue version. Perhaps try removing @vue/compiler-sfc and try again?

**Comment by Massedil** [comment]
Thanks for your suggestion, I tried removing `@vue/compiler-sfc`, but the problem is still here : 
https://framagit.org/massedil/mobilizon/-/commit/0391e5b8b08a3f9be9b9c6039bc71f6fd19a2cfc

**Comment by mlmoravek** [comment]
I'm sorry, I don't think I can help you :/
The sandbox example and my examples are working. You have a very large codebase with a lot of dependencies and I don't have the time to investigate your codebase further...

If anyone else has the same problem, please let me know!

When you find your solution, let me know what the problem was.

**Comment by Massedil** [comment]
I looked again at this issue with the 0.8.12 update. 

The good news is that the problem seems solved now.

To maybe helps you understand why, I re-run my minimal code : 
- 0.8.9 => @update:modelValue DONT works
- 0.8.10 => @update:modelValue DONT works
- 0.8.11 => @update:modelValue works 
- 0.8.12 => @update:modelValue works 

So, in fact, the problem is solved by 0.8.11 (I have not tested 0.8.11 before and I can test it easily now with the minimal code without `DateTimePicker`) : 
https://github.com/oruga-ui/oruga/compare/v0.8.10...v0.8.11

I still don't understand why you can't reproduce. :/
